### PR TITLE
[V2] test cases

### PR DESCRIPTION
**Pull request by ajaymathur**

### CODE REVIEW

**Comment by JedWatson** [comment]
👍 looks good @ajaymathur

File: src/Select.js [hunk 1]
```diff
@@ -335,10 +335,14 @@ export default class Select extends Component<Props, State> {
   }
   onMenuClose() {
     this.onInputChange('', { action: 'menu-close' });
-    this.props.onMenuClose();
+    if (this.props.onMenuClose) {
```
Comment (ajaymathur) [inline src/Select.js]: added these guards to not beak the component in case these are not provided

File: src/Select.js [hunk 2]
```diff
@@ -335,10 +335,14 @@ export default class Select extends Component<Props, State> {
...
+      this.props.onMenuClose();
+    }
   }
   onInputChange(newValue: string, actionMeta: InputActionMeta) {
-    this.props.onInputChange(newValue, actionMeta);
+    if (this.props.onInputChange) {
```
Comment (ajaymathur) [inline src/Select.js]: added these guards to not beak the component in case these are not provided

File: src/__tests__/AsyncCreatable.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  const asyncCreatableWrapper = mount(<AsyncCreatable />);
+  expect(asyncCreatableWrapper.find(Select).exists()).toBeTruthy();
+});
+
+test('render decorated select with props passed', () => {
+  const asyncCreatableWrapper = mount(<AsyncCreatable className="foo"/>);
+  expect(asyncCreatableWrapper.find(Select).props().className).toBe('foo');
+});
+
+test('to show the create option in menu', () => {
```
Comment (gwyneplaine) [inline src/__tests__/AsyncCreatable.test.js]: Please adjust this case to account for the fact that we only want to show the create option element, after the options have been loaded asynchronously.

File: src/__tests__/StateManaged.test.js [hunk 4]
```diff
@@ -0,0 +1,612 @@
...
+import cases from 'jest-in-case';
+
+import { OPTIONS } from './constants';
+import Select from '../';
+import { components } from '../components';
+
+const { Control, Menu, MultiValue, Placeholder, Option } = components;
+
+const BASIC_PROPS = { options: OPTIONS, name: 'test-input-name' };
+
```
Comment (gwyneplaine) [inline src/__tests__/StateManaged.test.js]: Please add test cases for validating the expected behaviour of exposed controlled props: 
1. props.inputValue
2. props.menuIsOpen 
3. props.value

File: src/__tests__/AsyncCreatable.test.js (hunk 3 above)
Comment (ajaymathur) [reply src/__tests__/AsyncCreatable.test.js]: done

File: src/__tests__/StateManaged.test.js (hunk 4 above)
Comment (ajaymathur) [reply src/__tests__/StateManaged.test.js]: done

**Comment by ajaymathur** [comment]
@gwyneplaine PR is ready for review 🙂

**Pull request merged**